### DISCUSSION TITLE
adding a new attribute group to existing combinations not possible

### DISCUSSION THREAD

**Discussion by Prestaworks** [PrestaShop/PrestaShop#40153] [Ideas]
**Describe the bug**
If I try to add a new choice to existing product combinations, I have to recreate all combinations. Stats are lost for the sales and I have to reenter all price changes. 
In 1.6 this was possible to edit on the combianation.

**To Reproduce**
1. Go to BO, product page for one of the t-shirts in the default installation.
2. Click on combinations
3. Try to add color choice to the existing size choices, without losing all data.

**Additionnal information**
PrestaShop version:1.7.4.2
PHP version: 5.6

**Comment by khouloudbelguith** [comment]
Hi @Prestaworks,

Thanks for your report.
Yes, I manage to reproduce this issue with PS1.7.4.4 / PS1.7.5.0rc1, it could be an improvement issue.
In the PS1.6, we can modify the attributes of any combination.
So we can add any attribute or delete it.
![image](https://user-images.githubusercontent.com/16067358/49640840-d41da980-fa0e-11e8-8df4-a04f4468b2b5.png)
@marionf, @colinegin what do you think?
Thanks!

**Comment by colinegin** [comment]
Hello @Prestaworks , 

Thanks for your feedback ! 
Indeed, there are some features related to combinations that were dropped from the 1.6.
We will be reworking the whole product and the combination feature, so do not hesitate to share more ideas and feedbacks.

In the meantime, have you checked if there is any module on the [Addons marketplace](https://addons.prestashop.com) which provides this feature ?

Have a nice day !

**Comment by darioesposito** [comment]
> 
> 
> On Prestashop 1.7, we can not modify / add an attribute of an existing combination by the B.O. We have to delete the combination in question, to recreate it with the desired attribute(s) (value).
> PRB: combination ID Is Removed => Problem with modules that use combination IDs (or **_id_product_attribute_**)
> 
> To avoid this, to add / modify an attribute of an existing combination WITHOUT deleting the combination in question, it is necessary to edit it, in the BDD.
> 
> Before that, we need 2 info / id:
> 
>     * The ID of the value of the attribute we want to modify / add
> 
>     * The ID of the existing combination
> 
> 
> For the ID of the value of the attribute you want to modify / add, here is the procedure to follow:
> 
>     1. Log in to the B.O of the site
> 
>     2. Go to **_Catalog> Attributes and Features_** and click on the attribute you want to modify / add (Example: **_Size_**)
> 
>     3. Locate the ID corresponding to the value of the attribute you want to modify / add on the existing combination.
>        For example, you want to add _Size S_ to a combination. By clicking on the **_Size_** attribute, you mark that the ID of size S is **_28_**
> 
> 
> If the value is not created, create it and find its ID
> Do the same for each additional attribute you want to modify / add
> 
> For the ID of the existing combination:
> 
>     1. Go to **_Catalog> Products_** and search for the product whose combination (s) you want to change
> 
>     2. Click on the pencil (on the right) to modify the product
> 
>     3. Go to the tabs **_Combinations_**
> 
>     4. Skip (not click) the mouse on the pencil to the right of the combination of which you want to modify / add an attribute
> 
>     5. At the bottom left of the window, there is a URL that appears. Find / hold the last digit (xxx), after "._.. combination_form_**xxx**_". For example: "_... combination_form_**_91_**"
> 
> 
> Make the modifications / additions in the database:
> 
>     1. Connect to the database via PHPMyAdmin
> 
>     2. Go to the table **_ps_product_attribute_combination_**
> 
> 
> If we want to **ADD an attribute** to an existing combination:
> 3. Click on **_Insert_**
> 4. In the **_id_attribute_** column, enter the ID of the attribute value (in our example **_28_**)
> 5. In the **_id_product_attribute_** column, enter the combination ID (in our example **_91_**)
> 6. Click on **_Run_**
> If you want to add other attributes to a combination, you will have to repeat the operation with a different attribute value ID each time (you must first have identified / retained the ID of the attribute value) but with the same combination ID.
> 
> If you want to **_MODIFY an attribute_** to an existing combination:
> 3. Click on **_Search_**
> 4. In the **_id_product_attribute column_**, enter the combination ID (in our example **_91_**)
> 5. Click on **_Run_**
> 6. Click **_Edit_** on the same line as the attribute to edit
> 7. In the id_attribute column, enter the ID of the attribute value (in our example **_28_**)
> 8. Click on **_Run_**
> 
> Verification:
> 
>     1. Return to **_B.O> Catalog> Products_** and search for the product whose combination (s) you want to change
> 
>     2. Click on the pencil (on the right) to modify the product
> 
>     3. Go to the tabs **_Combinations_**
>        (If you're already there, just refresh the page)
> 
>     4. Check that the changes / additions have been made
> 
> 
> I hope this will help other people who have this problem while waiting for Prestashop crew to fix this bug

Hi, 

Your solution work, but it's crazy thinking do stuff like that for an entire catalouge of products...

I hope they fix soon

Thanks anyway

**Comment by gh-hichem** [comment]
> > On Prestashop 1.7, we can not modify / add an attribute of an existing combination by the B.O. We have to delete the combination in question, to recreate it with the desired attribute(s) (value).
> > PRB: combination ID Is Removed => Problem with modules that use combination IDs (or **_id_product_attribute_**)
> > To avoid this, to add / modify an attribute of an existing combination WITHOUT deleting the combination in question, it is necessary to edit it, in the BDD.
> > Before that, we need 2 info / id:
> > ```
> > * The ID of the value of the attribute we want to modify / add
> > 
> > * The ID of the existing combination
> > ```
> > 
> > 
> > For the ID of the value of the attribute you want to modify / add, here is the procedure to follow:
> > ```
> > 1. Log in to the B.O of the site
> > 
> > 2. Go to **_Catalog> Attributes and Features_** and click on the attribute you want to modify / add (Example: **_Size_**)
> > 
> > 3. Locate the ID corresponding to the value of the attribute you want to modify / add on the existing combination.
> >    For example, you want to add _Size S_ to a combination. By clicking on the **_Size_** attribute, you mark that the ID of size S is **_28_**
> > ```
> > 
> > 
> > If the value is not created, create it and find its ID
> > Do the same for each additional attribute you want to modify / add
> > For the ID of the existing combination:
> > ```
> > 1. Go to **_Catalog> Products_** and search for the product whose combination (s) you want to change
> > 
> > 2. Click on the pencil (on the right) to modify the product
> > 
> > 3. Go to the tabs **_Combinations_**
> > 
> > 4. Skip (not click) the mouse on the pencil to the right of the combination of which you want to modify / add an attribute
> > 
> > 5. At the bottom left of the window, there is a URL that appears. Find / hold the last digit (xxx), after "._.. combination_form_**xxx**_". For example: "_... combination_form_**_91_**"
> > ```
> > 
> > 
> > Make the modifications / additions in the database:
> > ```
> > 1. Connect to the database via PHPMyAdmin
> > 
> > 2. Go to the table **_ps_product_attribute_combination_**
> > ```
> > 
> > 
> > If we want to **ADD an attribute** to an existing combination:
> > 3. Click on **_Insert_**
> > 4. In the **_id_attribute_** column, enter the ID of the attribute value (in our example **_28_**)
> > 5. In the **_id_product_attribute_** column, enter the combination ID (in our example **_91_**)
> > 6. Click on **_Run_**
> > If you want to add other attributes to a combination, you will have to repeat the operation with a different attribute value ID each time (you must first have identified / retained the ID of the attribute value) but with the same combination ID.
> > If you want to **_MODIFY an attribute_** to an existing combination:
> > 3. Click on **_Search_**
> > 4. In the **_id_product_attribute column_**, enter the combination ID (in our example **_91_**)
> > 5. Click on **_Run_**
> > 6. Click **_Edit_** on the same line as the attribute to edit
> > 7. In the id_attribute column, enter the ID of the attribute value (in our example **_28_**)
> > 8. Click on **_Run_**
> > Verification:
> > ```
> > 1. Return to **_B.O> Catalog> Products_** and search for the product whose combination (s) you want to change
> > 
> > 2. Click on the pencil (on the right) to modify the product
> > 
> > 3. Go to the tabs **_Combinations_**
> >    (If you're already there, just refresh the page)
> > 
> > 4. Check that the changes / additions have been made
> > ```
> > 
> > 
> > I hope this will help other people who have this problem while waiting for Prestashop crew to fix this bug
> 
> Hi,
> 
> Your solution work, but it's crazy thinking do stuff like that for an entire catalouge of products...
> 
> I hope they fix soon
> 
> Thanks anyway

i tried this, and it only works for one of the "IDs of the attribute value".
for example: 
if i'm trying to add three "ID of the attribute value" to the "id combination attribute" it only show the first one and not the others. and when i do it from the backoffice it works fine
